File: 김채린/추가문제/백준/N과M(2).java
import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.util.Arrays;

class Main {
    static StringBuilder result = new StringBuilder();
    private static void NAndM(int[] nums, int[] array, int n, int count) {
        if (count == nums[1]) {
            for (int i : array) {
                result.append(i).append(" ");
            }
            result.append("\n");
            return;
        }

        for (int i = n; i <= nums[0]; i++) {
            array[count] = i;
            NAndM(nums, array,i+1,count+1);
        }
    }
    public static void main(String[] args) throws Exception {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int[] nums = Arrays.stream(br.readLine().split(" ")).mapToInt(Integer::parseInt).toArray();
        int[] array = new int[nums[1]];
        NAndM(nums, array,1, 0);
        System.out.println(result);
    }
}

File: 김채린/추가문제/25일차/N과 M (2).kt
import java.io.*

val s = StringBuilder()
fun nAndM2(nums:List<Int>, array: Array<Int>, num:Int, count:Int) {
    if(count==nums[1]){
        for (i in array){
        s.append(i).append(" ")
        }
        s.append("\n")
        return
    }

    for (i in num..nums[0]){
            array[count] = i
            nAndM2(nums, array, i + 1, count+1)
    }
}

fun main(){
    val br = BufferedReader(System.`in`.reader())
    val nums = br.readLine().split(" ").map { it.toInt() }
    val array = Array(nums[1]){0}
    nAndM2(nums, array, 1, 0)
    println(s.toString())
}
